### PR TITLE
RELEASE: Updated news for v1.14.x

### PR DESCRIPTION
**Pull request by brminich**
## What
- Align news with the latest releases
- Add news for ucx v1.14

## Why ?
preparation for ucx v1.14 release

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -11,6 +11,333 @@
 ### Features:
 ### Bugfixes:
 
+## 1.14.0 (January 12, 2023)
+### Features:
+#### Core
+#### UCP
+* Added API for querying of transport and device names on endpoint
+* Added datatype query API
```
Comment (yosefe) [inline NEWS:19]: Added API for querying datatype object

File: NEWS [hunk 2]
```diff
@@ -11,6 +11,333 @@
...
+* Added new API ucp_memh_pack()
+* Added API to support exported memory handles (no implementation yet)
```
Comment (yosefe) [inline NEWS:21]: Unite the 2 lines: Added API for exporting and importing memory keys

File: NEWS [hunk 3]
```diff
@@ -11,6 +11,333 @@
...
+* Added support for non persistent header in AM API
```
Comment (yosefe) [inline NEWS:22]: Added support for non-persistent active message header

File: NEWS [hunk 4]
```diff
@@ -11,6 +11,333 @@
...
+* Added performance tree infrastructure for protocols v2
```
Comment (yosefe) [inline NEWS:23]: remove this. next line already captures it

File: NEWS [hunk 5]
```diff
@@ -11,6 +11,333 @@
...
+* Added infrastructure to print protocols v2 performance
+* Multiple performance improvements for protocols v2
+* Added support for non-contiguous datatypes in AM rendezvous protocol
+* Added support for non-contiguous datatypes for rendezvous protocols v2
```
Comment (yosefe) [inline NEWS:27]: unite

File: NEWS [hunk 6]
```diff
@@ -11,6 +11,333 @@
...
+* Added support for reset and abort request in protocols v2
+* Added rendezvous support for TAG sync send API in protocols v2
```
Comment (yosefe) [inline NEWS:29]: seems minor, remove

File: NEWS [hunk 7]
```diff
@@ -11,6 +11,333 @@
...
+* Added support for user memory handles in RMA API
+* Added multi-rail support for RMA API in protocols v2
+* Added support for up to 16 different lanes per endpoint
+* Added support for dmabuf memory registration in protocols v2
+* Added statistics to protocols v2
```
Comment (yosefe) [inline NEWS:34]: remove

File: NEWS [hunk 8]
```diff
@@ -11,6 +11,333 @@
...
+* Added strong fence mode for ucp_worker_fence() API
+* Added counter for pending connection requests on listener
```
Comment (yosefe) [inline NEWS:36]: minor, remove

File: NEWS [hunk 9]
```diff
@@ -11,6 +11,333 @@
...
+#### UCT
+* Added new uct_md_mem_attach() API to support exported memory handles
+* Added remote completion mode for endpoint flush (via new flag)
+* Added support for dmabuf registration
+* Added new uct_ep_connect_to_ep_v2() API
+* Added new uct_mem_reg_v2() API
+* Added new uct_md_query_v2() API
+#### RDMA CORE (IB, ROCE, etc.)
+* Added ECE(enhanced connection establishment) support for RC and DC transports
```
Comment (yosefe) [inline NEWS:45]: space before (

File: NEWS [hunk 10]
```diff
@@ -11,6 +11,333 @@
...
+* Added support for hardware DCS in DC transport
+* Added UD iface and endpoint resource information to VFS
+* Added CQ creation via DEVX API
+* Removed support for accelerated IB transports over experimental verbs
+#### UCS
+* Added support for auto-correction of user environment variables
+* Improved mpool API (added grow factor for number of elements in a chunk)
```
Comment (yosefe) [inline NEWS:52]: minor; remove

File: NEWS [hunk 11]
```diff
@@ -11,6 +11,333 @@
...
+* Added API to remove last element from array (ucs_array_pop_back)
```
Comment (yosefe) [inline NEWS:53]: minor; remove

File: NEWS [hunk 12]
```diff
@@ -11,6 +11,333 @@
...
+* Added support for IPv6 loopback address in TCP transport
```
Comment (yosefe) [inline NEWS:54]: why is this UCS and not UCT?

File: NEWS [hunk 13]
```diff
@@ -11,6 +11,333 @@
...
+#### UCM
+* Implemented CUDA bistro hooks for aarch64 (to enable memory cache on this platform)
+* Added support for CUDA virtual/stream-ordered memory with cudaMallocAsync
+#### GPU (CUDA, ROCM)
+* Implemented uct_iface_estimate_perf() function for ROCM
+* Removed obsoleted ROCM gdr transport
+* Added support for hsa async_copy for short operations in ROCM
+* Added memory allocation functions in ROCM
+* Added support for rcache parameters in ROCM copy transport
```
Comment (yosefe) [inline NEWS:63]: minor; remove

File: NEWS [hunk 14]
```diff
@@ -11,6 +11,333 @@
...
+#### Java
+* Added methods for ucp_worker_arm() and ucp_worker_get_efd()
```
Comment (yosefe) [inline NEWS:65]: I thought it already existed?

File: NEWS [hunk 15]
```diff
@@ -11,6 +11,333 @@
...
+#### Documentation
+* Added FAQ for using pkg-config tool to build applications with UCX
+* Added FAQ for PTX error with CUDA
```
Comment (yosefe) [inline NEWS:68]: remvoe

File: NEWS [hunk 16]
```diff
@@ -11,6 +11,333 @@
...
+#### Tests
+* Added test for CQE zipping
```
Comment (yosefe) [inline NEWS:70]: remove

File: NEWS [hunk 17]
```diff
@@ -11,6 +11,333 @@
...
+* Added prints of latency per connection in io_demo
+#### Tools
+* Added runtime library version to the 'ucx_info -v' output
+* Added support for memory types in ucx_info
+#### Build
+* Align all copyright headers with Nvidia standard
```
Comment (yosefe) [inline NEWS:76]: remove

File: NEWS [hunk 18]
```diff
@@ -11,6 +11,333 @@
...
+* Fixed bandwidth calculation for RoCE LAGs
+* Fixed port counters setting for DEVX QPs
+* Fixed compile errors on SLES sp3
+* Removed errors during md open in case of strict memlock limit
+#### UCS
+* Removed async_max_events limit (e.g. to support many concurrent TCP connections)
+* Updated memory wc flush using DGH hint for ARM platform
+* Fixed deprecation warnings because of <sys/fcntl.h> includes
+* Added default bandwidth value for ZHAOXIN CPU
+* Fixed logger to correctly print a pid
```
Comment (yosefe) [inline NEWS:126]: minor, remove

File: NEWS [hunk 19]
```diff
@@ -11,6 +11,333 @@
...
+* Fixed io_demo build on Ubuntu 20.04
+* Multiple improvements and fixes in UCP AM tests
+* Fixed connect_and_send_recv() in test_ucp_sockaddr
+* Fixed crash in test_ucp_sockaddr when RDMACM fails to resolve the peer address
+* Fixed UCT MD memory invalidation test (removed atomic access)
+* Multiple improvements/fixes to reduce overall testing time
+* Fixed gtest compilation error with -O1 optimization flag
+* Fixed memhooks test to load UCM first
+#### Tools
+* Replaced deprecated communication routines with *_nbx() APIs in ucp_hello_world
```
Comment (yosefe) [inline NEWS:152]: this is Tests

File: NEWS (hunk 12 above)
Comment (brminich) [reply NEWS:54]: changes are in src/ucs/sys/sock*

File: NEWS (hunk 14 above)
Comment (brminich) [reply NEWS:65]: it describes #8453

File: NEWS [hunk 20]
```diff
@@ -11,6 +11,333 @@
...
+#### Tools
+* Added runtime library version to the 'ucx_info -v' output
+* Added support for memory types in ucx_info
+#### Build
+* Align all copyright headers with Nvidia standard
+#### CI
+* Added io_demo testing with multiple HCAs
+* Added gtests with protocols v2
+* Added testing on Ubuntu 22.04 and 22.10 containers
+* Added tests for wire compatibility
```
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:81]: all CI section or just the last line?

File: NEWS [hunk 21]
```diff
@@ -11,6 +11,333 @@
...
+* Fixed copying of UCP context configuration (avoiding duplication of config strings)
+* Fixed worker iface initialization taking atomic caps into account
+* Fixed UCP AM max payload value calculation for protocols v2
+* Fixed deadlock in rcache when UCX_LOG_LEVEL set to debug
+* Fixed lanes weight calculation in rendezvous protocol v2
+* Fixed user memory handle support in rendezvous protocol
+* Fixed message split in rendezvous protocol to avoid having very small chunks
+* Improved performance estimations for protocols v2
+* Fixed receive descriptors leak in UCP AM rendezvous
+* Moved RKEY_PTR flag from memory domain to comnponent
```
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:99]: this is some kind of API change
remove anyway?

File: NEWS [hunk 22]
```diff
@@ -11,6 +11,333 @@
...
+* Aligned ucp_worker_fence() description with OSHMEM standard
+#### Tests
+* Fixed io_demo build on Ubuntu 20.04
+* Multiple improvements and fixes in UCP AM tests
+* Fixed connect_and_send_recv() in test_ucp_sockaddr
+* Fixed crash in test_ucp_sockaddr when RDMACM fails to resolve the peer address
+* Fixed UCT MD memory invalidation test (removed atomic access)
+* Multiple improvements/fixes to reduce overall testing time
+* Fixed gtest compilation error with -O1 optimization flag
+* Fixed memhooks test to load UCM first
```
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:150]: instead of the whole section?

File: NEWS (hunk 19 above)
Comment (brminich) [reply NEWS:152]: for 1.12 similar change is reported in `Core`. 
Also we mentioned `examples` in `Tools` section before

**Comment by brminich** [comment]
@shamisp, pls review as well

File: NEWS (hunk 12 above)
Comment (yosefe) [reply NEWS:54]: but it's for tcp transport

File: NEWS (hunk 14 above)
Comment (yosefe) [reply NEWS:65]: ok

File: NEWS (hunk 20 above)
[the remark this answers is not in the document]
Comment (yosefe) [reply NEWS:81]: all ci
it's not relevant for users

File: NEWS (hunk 21 above)
[the remark this answers is not in the document]
Comment (yosefe) [reply NEWS:99]: it's too minor change in uct

File: NEWS (hunk 22 above)
[the remark this answers is not in the document]
Comment (yosefe) [reply NEWS:150]: yes
users don't care so much about unit tests

File: NEWS (hunk 19 above)
Comment (yosefe) [reply NEWS:152]: it's definitely not Tools

File: NEWS [hunk 23]
```diff
@@ -11,6 +11,333 @@
 ### Features:
 ### Bugfixes:
 
+## 1.14.0 (January 12, 2023)
```
Comment (shamisp) [inline NEWS:14]: should be RC and not release ? @yosefe

File: NEWS [hunk 24]
```diff
@@ -11,6 +11,333 @@
...
+* Added new uct_md_mem_attach() API to support exported memory handles
+* Added remote completion mode for endpoint flush (via new flag)
+* Added support for dmabuf registration
+* Added new uct_ep_connect_to_ep_v2() API
+* Added new uct_mem_reg_v2() API
+* Added new uct_md_query_v2() API
+#### RDMA CORE (IB, ROCE, etc.)
+* Added ECE(enhanced connection establishment) support for RC and DC transports
+* Added support for hardware DCS in DC transport
+* Added UD iface and endpoint resource information to VFS
```
Comment (shamisp) [inline NEWS:47]: iface -> interface

File: NEWS [hunk 25]
```diff
@@ -11,6 +11,333 @@
...
+* Added CQ creation via DEVX API
+* Removed support for accelerated IB transports over experimental verbs
```
Comment (shamisp) [inline NEWS:49]: legacy support ?

File: NEWS [hunk 26]
```diff
@@ -11,6 +11,333 @@
...
+
+### Bugfixes
+#### UCP
+* Multiple fixes in keepalive protocol
+* Multiple fixes and improvements in UCP rcache flows
+* Fixed endpoints leak by disabling resolving remote endpoints in certain cases
+* Multiple fixes and cleanups in wireup protocol and lanes selection flows
+* Multiple fixes in protocols v2 infrastructure
+* Fixed copying of UCP context configuration (avoiding duplication of config strings)
+* Fixed worker iface initialization taking atomic caps into account
```
Comment (shamisp) [inline NEWS:91]: interface

**Review by shamisp — CHANGES_REQUESTED** [review]

File: NEWS (hunk 25 above)
Comment (brminich) [reply NEWS:49]: can you please elaborate?

File: NEWS (hunk 12 above)
Comment (brminich) [reply NEWS:54]: ok

File: NEWS (hunk 21 above)
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:99]: removed

File: NEWS (hunk 22 above)
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:150]: ok

File: NEWS (hunk 19 above)
Comment (brminich) [reply NEWS:152]: then, removed it, as Tests section contains just
`Multiple fixes and improvements in unit tests`

File: NEWS [hunk 27]
```diff
@@ -11,6 +11,333 @@
...
+#### Build
+* Fixed cuda detection (when cuda headers are not present, but nvml headers are)
+* Build Debian package with multi-thread support
+* Fixed configure warning by using POSIX compliant sh syntax
+* Multiple fixes for Debian package build
+#### CI
+* Removed test freezing in case of fatal error to avoid wasting worker resources
+* Fixed TCP port ranges to avoid overlapping between io_demo and client/server
+* Updated static library build to be a separate stage
+* Fixed RoCE failures on some machines (avoid cross-connecting RoCE devices)
```
[the remark this answers is not in the document]
Comment (brminich) [reply NEWS:163]: removed the whole CI section

File: NEWS (hunk 23 above)
Comment (shamisp) [reply NEWS]: @brminich we start with RC and remove RC from the release name just before official release.

File: NEWS (hunk 25 above)
Comment (yosefe) [reply NEWS:44]: "Removed support for accelerated IB transports over legacy experimental verbs"

**Review by yosefe — COMMENTED** [review]
LGTM besides the comments from Pasha

**Review by yosefe — APPROVED** [review]

**Comment by shamisp** [comment]
Looks good. Please squash.

**Comment by brminich** [comment]
@shamisp, sqashed, but there are still 2 commits:
- sync with prev 1.13.x releases   
- new updates for 1.14 release

**Review by yosefe — APPROVED** [review]

**Pull request merged**